alma/controllers/hook/StateHookController.php

```diff
@@ -43,38 +44,86 @@ final class StateHookController extends AdminHookController
     use OrderDataTrait;
 
     /**
+     * Execute refund or trigger payment on change state
      * @param $params
      *
      * @throws PrestaShopDatabaseException
      * @throws PrestaShopException
      */
     public function run($params)
     {
-        if (!Settings::isRefundEnabledByState()) {
-            return;
-        }
-
         $order = new Order($params['id_order']);
         $newStatus = $params['newOrderStatus'];
         if ($order->module !== 'alma' || !$order_payment = $this->getOrderPaymentOrFail($order)) {
             return;
         }
-
+        $alma = ClientHelper::defaultInstance();
+        if (!$alma) {
+            return;
+        }
         $id_payment = $order_payment->transaction_id;
-        if ($newStatus->id == Settings::getRefundState()) {
-            $alma = ClientHelper::defaultInstance();
-            if (!$alma) {
-                return;
+
+        try {
+            switch ($newStatus->id) {
+                case Settings::getRefundState():
+                    if (!Settings::isRefundEnabledByState()) {
+                        throw new NotPermittedBySettingsException('Refund option is not enabled');
+                    }
+                    $this->refund($alma, $id_payment, $order);
+                    break;
+                case Settings::getPaymentTriggerState():
+                    if (!Settings::isPaymentTriggerEnabledByState()) {
+                        throw new NotPermittedBySettingsException('Payment trigger option is not enabled');
+                    }
+                    $this->triggerPayment($alma, $id_payment, $order);
+                    break;
+                default :
+                    return;
             }
+        } catch (NotPermittedBySettingsException $e) {
```

Comment:
Exceptions should be thrown to allow an external call of the method to catch them.

Here, I think the usage is not right.

If a settings is enabled, do the action, if not, don't do it (KISS)
If you need to log that settings is not enabled, maybe you can do it directly into the `is...StateEnabled()` method ?

Reply:
Do you think the try catch is useless and that I should remove it?